toolkit/tools/pkg/imagecustomizerlib/artifactsinputoutput.go

```diff
@@ -95,15 +144,143 @@ func outputArtifacts(items []imagecustomizerapi.OutputArtifactsItemType, outputD
 		destPath := filepath.Join(outputDir, bootConfig.systemdBootBinary)
 		err := file.Copy(srcPath, destPath)
 		if err != nil {
-			return fmt.Errorf("failed to copy binary from (%s) to (%s):\n%w", srcPath, destPath, err)
+			return nil, fmt.Errorf("failed to copy binary from (%s) to (%s):\n%w", srcPath, destPath, err)
 		}
+
+		signedPath := prefix + replaceSuffix(filepath.Join(configOutputArtifactsPath, bootConfig.systemdBootBinary), ".efi", ".signed.efi")
+
+		outputArtifactsMetadata = append(outputArtifactsMetadata, OutputArtifactMetadata{
+			Partition:   partition,
+			Source:      signedPath,
+			Destination: filepath.Join("/EFI/systemd", bootConfig.systemdBootBinary),
+		})
 	}
 
 	err = systemBootPartitionMount.CleanClose()
+	if err != nil {
+		return nil, err
+	}
+
+	err = loopback.CleanClose()
+	if err != nil {
+		return nil, err
+	}
+
+	return outputArtifactsMetadata, nil
+}
+
+func replaceSuffix(input string, oldSuffix string, newSuffix string) string {
+	if !strings.HasSuffix(input, oldSuffix) {
+		return input
+	}
+	return strings.TrimSuffix(input, oldSuffix) + newSuffix
+}
+
+func writeInjectFilesYaml(metadata []OutputArtifactMetadata, baseConfigPath string) error {
+	type injectFileYaml struct {
```

Comment:
You have both `injectFileYaml` and `InjectFilesYaml` structs. Please remove `injectFileYaml`.